src/bindings/g_object/type.cr

```diff
@@ -22,6 +22,144 @@ module GObject
+    LibGObject.g_param_spec_boolean(name, nick, blurb, default, flags)
+  end
+
+  # :nodoc:
+  def self.create_param_spec(klass : Int8.class, name, nick, blurb, flags, *, min : Int8 = Int8::MIN, max : Int8 = Int8::MAX, default : Int8 = 0i8) : Void*
+    LibGObject.g_param_spec_char(name, nick, blurb, min, max, default, flags)
+  end
+
+  # :nodoc:
+  def self.create_param_spec(klass : UInt8.class, name, nick, blurb, flags, *, min : UInt8 = UInt8::MIN, max : UInt8 = UInt8::MAX, default : UInt8 = 0u8) : Void*
+    LibGObject.g_param_spec_uchar(name, nick, blurb, min, max, default, flags)
+  end
+
+  # :nodoc:
+  def self.create_param_spec(klass : Int32.class, name, nick, blurb, flags, *, min : Int32 = Int32::MIN, max : Int32 = Int32::MAX, default : Int32 = 0) : Void*
+    LibGObject.g_param_spec_int(name, nick, blurb, min, max, default, flags)
+  end
+
+  # :nodoc:
+  def self.create_param_spec(klass : UInt32.class, name, nick, blurb, flags, *, min : UInt32 = UInt32::MIN, max : UInt32 = UInt32::MAX, default : UInt32 = 0u32) : Void*
+    LibGObject.g_param_spec_uint(name, nick, blurb, min, max, default, flags)
+  end
+
+  # :nodoc:
+  def self.create_param_spec(klass : Int64.class, name, nick, blurb, flags, *, min : Int64 = Int64::MIN, max : Int64 = Int64::MAX, default : Int64 = 0i64) : Void*
+    LibGObject.g_param_spec_int64(name, nick, blurb, min, max, default, flags)
+  end
+
+  # :nodoc:
+  def self.create_param_spec(klass : UInt64.class, name, nick, blurb, flags, *, min : UInt64 = UInt64::MIN, max : UInt64 = UInt64::MAX, default : UInt64 = 0u64) : Void*
+    LibGObject.g_param_spec_uint64(name, nick, blurb, min, max, default, flags)
+  end
+
+  # :nodoc:
+  def self.create_param_spec(klass : Float32.class, name, nick, blurb, flags, *, min : Float32 = Float32::MIN, max : Float32 = Float32::MAX, default : Float32 = 0f32) : Void*
+    LibGObject.g_param_spec_float(name, nick, blurb, min, max, default, flags)
+  end
+
+  # :nodoc:
+  def self.create_param_spec(klass : Float64.class, name, nick, blurb, flags, *, min : Float64 = Float64::MIN, max : Float64 = Float64::MAX, default : Float64 = 0.0) : Void*
+    LibGObject.g_param_spec_double(name, nick, blurb, min, max, default, flags)
+  end
+
+  # :nodoc:
+  def self.create_param_spec(klass : GObject::Object?.class, name, nick, blurb, flags) : Void*
+    LibGObject.g_param_spec_object(name, nick, blurb, type_not_nil!(klass).g_type, flags)
+  end
+
+  # :nodoc:
+  def self.create_param_spec(klass : Enum.class, name, nick, blurb, flags, *, default : Enum? = nil) : Void*
+    if klass._is_flags_enum?
+      LibGObject.g_param_spec_flags(name, nick, blurb, klass.g_type, default || 0, flags)
+    else
+      LibGObject.g_param_spec_enum(name, nick, blurb, klass.g_type, default || 0, flags)
+    end
+  end
+
+  # :nodoc:
+  def self.fundamental_g_type(klass : String.class) : UInt64
+    TYPE_STRING
+  end
```

Comment:
In `type.cr` we define a `self.g_type` method for most of the types, I think we can use them.

Anyway, I think this is good to be merged anyway and minor improvements can land in future patches.

Awesome work!!